### PR TITLE
Document data verification script

### LINKED ISSUE #50: Extend the CI to verify the Data Feed IDs and Sponsor Wallet Addresses (dAPI management)

### PR DESCRIPTION
**Pull request by mcoetzee**
Follow up on #91 
Closes #50 

### What does this change?
- adds documentation for the `scripts/verify-data.js` script

### CODE REVIEW

File: scripts/verification/README.md [hunk 1]
```diff
@@ -0,0 +1,33 @@
+# Data Verification
+
+The `scripts/verify-data.js` script verifies all merkle tree data and has two distinct parts.
+
+## Part 1: Verify all merkle tree data (before `HashRegistry` checks)
+
+For each current merkle tree (`current-hash.json`) we assert that:
+
+- the `hash` value matches the constructed merkle tree root
+- the `signatures` keys match the hash signers exactly (from `hash-signers.json`)
```
Comment (acenolaza) [inline scripts/verification/README.md:10]: The order of the `keys` matching the order of the values in `hash-signers.json` is implicit by using "exactly", right?

File: scripts/verification/README.md [hunk 2]
```diff
@@ -0,0 +1,33 @@
...
+- all signatures are valid
+
+### dAPI Management Merkle tree
+
+Additional assertions include that:
+
+- all Data Feed IDs are derived from the dAPI Name and corresponding API Providers
+- all Sponsor Wallet Addresses are derived from the dAPI Name
```
Comment (acenolaza) [inline scripts/verification/README.md:18]: Nit: 
```suggestion
- all Sponsor Wallet addresses are derived from the dAPI Name
```

**Review by acenolaza — APPROVED** [review]
LGTM 👍🏻

File: scripts/verification/README.md (hunk 1 above)
Comment (mcoetzee) [reply scripts/verification/README.md:10]: Yes that's right. Do you think I should be more explicit wrt the order of the keys?

Comment (mcoetzee) [reply scripts/verification/README.md]: I updated it to be a bit more explicit 👍

**Pull request merged**